src/shared/bridge.js

```diff
@@ -30,12 +33,18 @@ export default class Bridge {
    * The created channel is added to the list of channels which `call`
    * and `on` send and receive messages over.
    *
-   * @param {Window} source - The source window.
+   * @param {Window|MessagePort} source - The message source.
    * @param {string} origin - The origin of the document in `source`.
    * @param {string} token
    * @return {RPC} - Channel for communicating with the window.
+   *
+   * @deprecated
    */
   createChannel(source, origin, token) {
```

Comment:
There are currently two different public APIs for creating a channel from a `MessagePort` - calling `createChannel` or calling `createChannelFromPort` directly. This seems redundant to me. I can suggest a few different ways to rectify this:

- Package up the data needed to create a channel for a window into one object, then make `createChannel` accept either that object or a `MessagePort`, and only have `createChannel` as the public API
- Have two separate public methods, `createChannelForWindow` and `createChannelForPort`, which work with a Window and MessagePort respectively. Internally these could delegate to a shared helper to do most of the work

Reply:
Yes, I can do something along those lines. The two implementations are 95% similar, that's why I thought it was OK to have both of them, since one of them is marked `@deprecated` and will be removed.